src/os/bluestore/BlueStore.cc

```diff
@@ -14619,6 +14619,16 @@ void BlueStore::_do_truncate(
     o->extent_map.fault_range(db, offset, length);
     o->extent_map.punch_hole(c, offset, length, &wctx.old_extents);
     o->extent_map.dirty_range(offset, length);
+
+    if (bdev->is_smr()) {
+      o->onode.size = offset;
```

Comment:
Honestly I don't see any rationales to have this line here. zoned_note_truncated_object saves an onode reference hence no much need to update the size before - it will get proper instance anyway.

Other than that - LGTM

Reply:
Makes sense. Updated.
Thank you.